### ISSUE TITLE
Array.prototype.push typing

### ISSUE THREAD

**Issue by ak239** [google/closure-compiler#1611]
Hello,

I write following code:
```js
var items = /** @type {!Array.<!WebInspector.ToolbarItem>} */([]);
items.push(/** @type {!Element} */(parentElement));
```
And it was compiled by closure compiler without error.
It it issue or feature?

**Comment by Dominator008** [comment]
What's the type of `WebInspector.ToolbarItem`?

**Comment by ak239** [comment]
It extends type with following declaration:
/**
 * @constructor
 * @implements {WebInspector.EventTarget}
 */

And compiler doesn't produce error for more simple case like:
var items = /** @type {!Array.<number>} */([]);
items.push(/** @type {!Element} */(parentElement));

Maybe the root of issue is our outdated version of closure compiler. We roll it around half year ago.

**Comment by Dominator008** [comment]
I'm closing this since i can't reproduce it on the current master. Please update the compiler and try again.
